docs/toggles.rst

```diff
@@ -1,47 +1,35 @@
-
-admin UI. These are defined by adding a new ``StaticToggle`` in this file. See ``PredictablyRandomToggle``
-
-and ``DynamicallyPredictablyRandomToggle`` if you need a toggle to be defined for a random subset
-
-of users.
+For self-hosted environments, plugins typically have a higher support burden and require directly reading the code to provide support. A smaller percentage of CommCare’s open source developer community typically has knowledge on any given plugin. If you are enabling plugins in your local environment, please make sure you have sufficient engineering expertise to be able to read direct code-level documentation. Plugins can be managed through the this `admin <UI https://www.commcarehq.org/hq/flags/?__hstc=240960668.9cbe5668b14d773d52583aa6ddd44168.1491228590522.1491493477971.1491498579156.19&__hssc=240960668.47.1491498579156&__hsfp=2069207928>`_.
 
-Namespaces define the type of access granted. NAMESPACE_DOMAIN allows the toggle to be enabled
+The CommCare Community of Practice urges all plugin maintainers to follow our best practices for `documentation <(https://commcare-hq.readthedocs.io/documenting.html>`_. Each commit should include a description of the functionality and links to relevant tickets.
 
-for individual project spaces. NAMESPACE_USER allows the toggle to be enabled for individual users,
+Plugins allow limiting access to a set of functionality.
 
-with the functionality visible to only that user but on any project space they visit.
-
-NAMESPACE_DOMAIN is preferred for most flags, because it can be confusing for different users
+They are implemented as a couchdb-backed django app, designed to be *simple* and *fast* (automatically cached).
 
-to experience different behavior. Domain-based flags are like a lightweight privilege that's
+Most plugins are configured by manually adding individual users or domains in the plugins admin UI. These are defined by adding a new ``StaticToggle`` in this file. See ``PredictablyRandomToggle`` and ``DynamicallyPredictablyRandomToggle`` if you need a plugin to be defined for a random subset of users.
 
-independent of a software plan. User-based flags are more like a lightweight permission that's
+Namespaces define the type of access granted. NAMESPACE_DOMAIN allows the plugin to be enabled for individual project spaces. NAMESPACE_USER allows the plugin to be enabled for individual users, with the functionality visible to only that user but on any project space they visit.
 
-independent of user roles (and therefore also independent of domain).
+NAMESPACE_DOMAIN is preferred for most flags, because it can be confusing for different users to experience different behavior. Domain-based flags are like a lightweight privilege that's independent of a software plan. User-based flags are more like a lightweight permission that's independent of user roles (and therefore also independent of domain).
```

Comment:
The language here about software plans is good, by the way. It's fine for us to refer to plans and paid versions, etc, since the Open Source code does define and use those concepts, we should just avoid referring to our particular support and subscription plans.